src/workerd/api/worker-rpc.h

```diff
@@ -259,7 +316,39 @@ class JsRpcStub: public JsRpcClientProvider {
   JSG_SERIALIZABLE(rpc::SerializationTag::JS_RPC_STUB);
 
 private:
-  IoOwn<rpc::JsRpcTarget::Client> capnpClient;
+  // Nulled out upon dispose().
+  kj::Maybe<IoOwn<rpc::JsRpcTarget::Client>> capnpClient;
+
+  kj::Maybe<RpcStubDisposalGroup&> disposalGroup;
+  kj::ListLink<JsRpcStub> disposalGroupLink;
+
+  friend class RpcStubDisposalGroup;
+};
+
+class RpcStubDisposalGroup {
```

Comment:
non-blocking: I wonder if this is a pattern we might end up re-using for other disposable things.. might be worth abstracting a bit to have a generic `DisposalGroup` type.

Reply:
Eh I don't want to create premature abstractions. Let's wait until we have a use case.